### PR TITLE
Use ubi for container images

### PR DESCRIPTION
**Pull request by dbaker-rh**
Leverage UBI to build and run the container image.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,17 +1,17 @@
 # Build the manager binary
-FROM golang:1.16 as builder
+FROM registry.access.redhat.com/ubi8/go-toolset:latest as builder
```
Comment (frobware) [inline Dockerfile:2]: Shouldn't this be similar (if not identical) to the way we build (for example) the cluster-ingress-operator?

```
FROM registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.16-openshift-4.10 AS builder
```

Taken from: https://github.com/openshift/cluster-ingress-operator/commit/c22568aac5be683cd0dbf1eba43522a9645455ca

Comment (dbaker-rh) [reply Dockerfile:2]: This Dockerfile is for end-users building manually (e.g. in the README.md manual steps for "make image").

In this repo, there is Dockerfile.openshift which tracks that change for the 4.9 ... 4.10 changes, and which uses the ART images.

**Comment by frobware** [comment]
/lgtm

based on the understanding that `Dockerfile.openshift` is for OpenShift CI and this addition doesn't affect CI builds. Also seems like a ~200MB image size saving. Makes me wonder why `Dockerfile.openshift` cannot produce a similar size image.

File: Dockerfile [hunk 2]
```diff
@@ -1,17 +1,17 @@
...
 
-WORKDIR /workspace
+WORKDIR /opt/app-root/src
 COPY . .
 
 # Build
 RUN make build-operator
 
 # Use distroless as minimal base image to package the manager binary
 # Refer to https://github.com/GoogleContainerTools/distroless for more details
```
Comment (dbaker-rh) [inline Dockerfile]: I just re-reviewed, and see this -- I need to update my commit to remove this comment now we're using UBI instead.

**Comment by dbaker-rh** [comment]
@frobware - can you re-lgtm?  I pushed a new commit to fix the comments.

**Comment by frobware** [comment]
> @frobware - can you re-lgtm? I pushed a new commit to fix the comments.

Sure, but you may as well squash first.

**Pull request merged**